### ISSUE TITLE
Problems with latest version (2021.7 Early Adopter)

### ISSUE THREAD

**Issue by soffensive** [PortSwigger/viewstate-editor#1]
Can you please check if there is a compatibility issue with the latest available Burp version 2021.7? I cannot make it work in this version, but the stable version still works apparently.
Sorry for the lack of deep investigation.

Generally, the problem is that the extension does not appear in the Request / Response viewer of Burp.

**Comment by Regala** [comment]
Can confirm it doesn't work in early adopter.

Potential interesting stack trace:
```
	at burp.fw4.addNotify(Unknown Source)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addImpl(Container.java:1146)
	at java.desktop/javax.swing.JTabbedPane.insertTab(JTabbedPane.java:754)
	at burp.d6t.a(Unknown Source)
	at burp.d6t.insertTab(Unknown Source)
	at java.desktop/javax.swing.JTabbedPane.addTab(JTabbedPane.java:828)
	at burp.bor.a(Unknown Source)
	at burp.f1g.c(Unknown Source)
	at burp.f1g.b(Unknown Source)
	at burp.zz.a(Unknown Source)
	at burp.abc.addSuiteTab(Unknown Source)
	at burp.at3.addSuiteTab(Unknown Source)
	at burp.u0.addSuiteTab(Unknown Source)
	at burp.BurpExtender$1.run(BurpExtender.java:127)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
java.lang.NullPointerException: Request cannot be null
	at burp.du5.analyzeRequest(Unknown Source)
	at burp.du5.analyzeRequest(Unknown Source)
	at burp.i5l.analyzeRequest(Unknown Source)
	at burp.ViewStateTab.isEnabled(ViewStateTab.java:84)
	at burp.bp3.a(Unknown Source)
	at burp.fht.a(Unknown Source)
	at burp.bn4.c(Unknown Source)
	at burp.fw4.a(Unknown Source)
	at burp.fw4.a(Unknown Source)
	at burp.f1g.b(Unknown Source)
	at burp.b2p.b(Unknown Source)
	at burp.fw4.addNotify(Unknown Source)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addImpl(Container.java:1146)
	at java.desktop/javax.swing.JTabbedPane.insertTab(JTabbedPane.java:754)
	at burp.d6t.a(Unknown Source)
	at burp.d6t.insertTab(Unknown Source)
	at java.desktop/javax.swing.JTabbedPane.addTab(JTabbedPane.java:828)
	at burp.bor.a(Unknown Source)
	at burp.f1g.c(Unknown Source)
	at burp.f1g.b(Unknown Source)
	at burp.zz.a(Unknown Source)
	at burp.abc.addSuiteTab(Unknown Source)
	at burp.at3.addSuiteTab(Unknown Source)
	at burp.u0.addSuiteTab(Unknown Source)
	at burp.BurpExtender$1.run(BurpExtender.java:127)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
java.lang.NullPointerException: Request cannot be null
	at burp.bp3.a(Unknown Source)
	at burp.fht.a(Unknown Source)
	at burp.bn4.c(Unknown Source)
	at burp.fw4.a(Unknown Source)
	at burp.fw4.a(Unknown Source)
	at burp.f1g.b(Unknown Source)
	at burp.b2p.b(Unknown Source)
	at burp.fw4.addNotify(Unknown Source)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4783)
	at java.desktop/java.awt.Container.addImpl(Container.java:1146)
	at java.desktop/javax.swing.JTabbedPane.insertTab(JTabbedPane.java:754)
	at burp.d6t.a(Unknown Source)
	at burp.d6t.insertTab(Unknown Source)
	at java.desktop/javax.swing.JTabbedPane.addTab(JTabbedPane.java:828)
	at burp.bor.a(Unknown Source)
	at burp.f1g.c(Unknown Source)
	at burp.f1g.b(Unknown Source)
	at burp.zz.a(Unknown Source)
	at burp.abc.addSuiteTab(Unknown Source)
	at burp.at3.addSuiteTab(Unknown Source)
	at burp.u0.addSuiteTab(Unknown Source)
	at burp.BurpExtender$1.run(BurpExtender.java:127)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

^ this might or not be related i.e. the exception can be just normally be thrown if the extension tries to access the response of a request and not the request itself. Regardless if by `addSuiteTab`  it means add a new tab like an extension would (not in the request/response editor) then that is def a problem.

**Comment by irsdl** [comment]
I have the same error in 2021.7.1. 
Surprisingly even viewing the sitemap in Burp causes a similar error (if you open burp using the command line to see the errors). So I am not 100% sure whether this is Java or Burp that has changed but this extension does not work now and needs updating.

**Comment by irsdl** [comment]
I confirm this issue has been resolved in Burp 2021.7.2

**Comment by soffensive** [comment]
Thanks for the update, I checked again with the latest version as of now: v2021.7.2-8935 (Early Adopter)

The bug seems to have been fixed
